source/simulator/solver.cc

```diff
@@ -190,9 +190,9 @@ namespace aspect
          * @param S_block_tolerance The tolerance for the CG solver which computes
          *     the inverse of the S block (Schur complement matrix).
          **/
-        BlockSchurPreconditioner (const LinearAlgebra::BlockSparseMatrix  &S,
-                                  const LinearAlgebra::BlockSparseMatrix  &Spre,
-                                  const PreconditionerMp                     &Mppreconditioner,
+        BlockSchurPreconditioner (const LinearAlgebra::BlockSparseMatrix     &S,
+                                  const LinearAlgebra::BlockSparseMatrix     &Spre,
+                                  const SInvOperator                         &S_inverse_operator,
                                   const PreconditionerA                      &Apreconditioner,
```

Comment:
That would then also apply to the naming of the two variables of the types in question.

Reply:
PreconditionerA is itself a preconditioner that is involved in a solve with the A block, so  I think it would make sense to still describe it as a preconditioner, whereas SInvOperator is an approximation of the action of S^{-1}, which is only part of applying the Schur complement preconditioner.

Reply:
In other words: SInvOperator contains the MpPreconditioner.

I propose the following: We merge the PR and I will followup with a PR that also extracts the solve for A. Then the class will have two arguments ``AInvOperator`` and ``SInvOperator``.

Reply:
The way I see it is that a preconditioner *is* an approximation to the inverse of the matrix.

Reply:
Quang, can you rename ``PreconditionerA`` to ``AInvOperator`` for now? We will extract the solve later.